kernel/src/cpu/control_regs.rs

```diff
@@ -24,10 +24,14 @@ pub fn cr4_init() {
 
     cr4.insert(CR4Flags::PSE); // Enable Page Size Extensions
 
-    if cpu_has_pge() {
-        cr4.insert(CR4Flags::PGE); // Enable Global Pages
+    if !cpu_has_pge() {
+        // All processors that are capable of virtualization will support
+        // global page table entries, so there is no reason to support any
+        // processor that does not enumerate PGE capability.
+        panic!("CPU does not support PGE");
     }
```

Comment:
Minor style suggestion:

```suggestion
    // All processors that are capable of virtualization will support
    // global page table entries, so there is no reason to support any
    // processor that does not enumerate PGE capability.
    assert!(cpu_has_pge(), "CPU does not support PGE");
```

Reply:
I changed this to an assertion, but now I'm considering changing it back.  The assertion will only be performed in debug builds, but we want to prevent boot even in non-debug builds.  Doesn't that make `panic` more appropriate?

Reply:
Actually, I see that `assert` happens always and `debug_assert` is debug-only, so `assert` is appropriate here.

Reply:
Correct, `assert` is always compiled in.